skbio/io/format/fasta.py

```diff
@@ -640,22 +640,26 @@ def _fasta_sniffer(fh):
     #   not be validated but it probably isn't what the user wanted). Also, if
     #   we add QUAL as its own file format in the future, we wouldn't want the
     #   FASTA and QUAL sniffers to both positively identify a QUAL file.
-    if _too_many_blanks(fh, 5):
-        return False, {}
+    blanks, consumed = _too_many_blanks(fh, 5)
+    if blanks:
+        return False, {}, consumed
+    if not fh.seekable():
+        fh = itertools.chain(consumed, fh)
 
+    consumed = []
     num_records = 10
     empty = True
     try:
         parser = _parse_fasta_raw(fh, _sniffer_data_parser, FASTAFormatError)
-        for _ in zip(range(num_records), parser):
+        # Check 10 records, record is in the form of (index, data)
+        # Data is returned as consumed from the parser
+        for record in zip(range(num_records), parser):
+            consumed += record[1][3]
```

Comment:
I'm not really sure what this is doing or why, the comment doesn't add much information here

It may help to perform a tuple expansion within the for loop statement, but even then, it's not obvious why the third index of the data variable is important